src/main/java/org/kohsuke/github/GHRepositoryRule.java

```diff
@@ -0,0 +1,692 @@
+package org.kohsuke.github;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+
+/**
+ * Represents a repository rule.
+ */
+public class GHRepositoryRule {
+    private Type type;
+    private RulesetSourceType ruleset_source_type;
+    private String ruleset_source;
+    private long ruleset_id;
+    private Map<String, JsonNode> parameters;
+
+    /**
+     * Instantiates a new GH repository rule.
+     */
+    public GHRepositoryRule() {
+    }
+
+    /**
+     * Instantiates a new GH repository rule.
+     *
+     * @param type
+     *            the type
+     * @param ruleset_type_source
+     *            the ruleset type source
+     * @param ruleset_source
+     *            the ruleset source
+     * @param ruleset_id
+     *            the ruleset id
+     * @param parameters
+     *            the parameters
+     */
+    public GHRepositoryRule(Type type,
+            RulesetSourceType ruleset_type_source,
+            String ruleset_source,
+            long ruleset_id,
+            Map<String, JsonNode> parameters) {
+        this.type = type;
+        this.ruleset_source_type = ruleset_type_source;
+        this.ruleset_source = ruleset_source;
+        this.ruleset_id = ruleset_id;
+        this.parameters = new HashMap<>(parameters);
+    }
```

Comment:
Jackson doesn't require a constructor to populate the fields in a record. The lack of code coverage proves Jackson truly not using this. Remove it.